Articles

trendwithin opened the pull request:
Create a Blog template-
Use integration testing to verify an user experience through the creation, deletion
and editing of an article.

trendwithin committed:
Modify Gemfile Add test/test_helper

trendwithin committed:
Add Home Page, controllers- test clean

trendwithin committed:
Add First Test Case

trendwithin committed:
Add Articles Page

trendwithin committed:
Add Visiting Articles Page

trendwithin committed:
Verified Test

test/controllers/articles_controller.rb
test/feattures/visiting_the_article_test.rb

trendwithin committed:
Test Editing an Article

test/features/editing_an_article_test.rb passes

trendwithin committed:
Add Flash Notice For Create/Edit/Delete

trendwithin committed:
Deleting An Article

passed test for deleting an article

trendwithin committed:
Add Image to README

trendwithin committed:
Add Blog Screen Capture

trendwithin committed:
Modify README

sealocal commented on test/features/editing_an_article_test.rb:
```diff
@@ -0,0 +1,30 @@
+
+  #when I submit the form
+  click_on "Create Article"
+
```
You could DRY up this test by replacing lines 6 through 13 with one line:

``` ruby
 article = Article.create(title: 'Code Rails', body: 'This is how I learned to make web apps')
```

That change would make this test use the same approach you're using in `test/features/deleting_an_article_test.rb` (see [line 6](https://github.com/trendwithin/portfolio/blob/articles/test/features/deleting_an_article_test.rb#L6)).

trendwithin replied:
Agreed. That's my 6AM project tomorrow.  I'll be working on a different
branch and push it ASAP.
On Mar 6, 2015 6:23 PM, "Mike Taylor" notifications@github.com wrote:

> In test/features/editing_an_article_test.rb
> https://github.com/trendwithin/portfolio/pull/2#discussion_r25992205:
> 
> > @@ -0,0 +1,30 @@
> > +require "test_helper"
> > +
> > +feature "Ability to edit and modify a blog post" do
> > -  scenario "Clicking edit button brings up blog post" do
> >   +
> > -    visit articles_path
> > -    click_link 'New Article'
> > -    #given
> > -    fill_in "Title", with: "Code Rails"
> > -    fill_in "Body", with: "This is how I learned to make web apps"
> >   +
> > -  #when I submit the form
> > -  click_on "Create Article"
> >   +
> 
> You could DRY up this test by replacing lines 6 through 13 with one line:
> 
>  article = Article.create(title: 'Code Rails', body: 'This is how I learned to make web apps')
> 
> That change would make this test use the same approach you're using in
> test/features/deleting_an_article_test.rb (see line 6
> https://github.com/trendwithin/portfolio/blob/articles/test/features/deleting_an_article_test.rb#L6).
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/trendwithin/portfolio/pull/2/files#r25992205.

trendwithin committed:
Refactor Test Code-

Currently a test clash that needs to be
resolved and further refactoring required

trendwithin committed:
Refactor test cases

Resolve clash and further refactor required

trendwithin committed:
Continue to refactor test code

modify drivers

trendwithin committed:
Refactor Test- All Pass

trendwithin committed:
Remove files not requried

by the current implementation of the project
